domains/SignIn/index.js

```diff
@@ -86,6 +87,12 @@ const SignIn = ({ navigation }) => {
     I18n.locale = lang;
   };
 
+  // checks whether user is connected to internet, return true if connected, false otherwise
+  async function checkOnlineStatus() {
```

Comment:
great function. 

you should create a module named `offline`  in the `modules` folder and move functions related to offline data collection like this to it. `expo-network` imports shouldn't be in any of the forms